backend/src/main/java/com/woowacourse/friendogly/club/service/ClubQueryService.java

```diff
@@ -39,18 +44,23 @@ public List<FindSearchingClubResponse> findSearching(FindSearchingClubRequest re
                 .build();
 
         return clubRepository.findAll(spec).stream()
-                .map(club -> {
-                    List<String> overviewPetImages = clubMemberRepository.findAllByClubId(club.getId()).stream()
-                            .map(ClubMember::findOverviewPetImage)
-                            .toList();
-                    return new FindSearchingClubResponse(
-                            club,
-                            clubMemberRepository.countByClubId(club.getId()),
-                            overviewPetImages
-                    );
-                })
+                .map(club -> new FindSearchingClubResponse(
+                        club,
+                        clubMemberRepository.countByClubId(club.getId()),
+                        collectOverviewPetImages(club)
+                ))
                 .toList();
     }
+
+    private List<String> collectOverviewPetImages(Club club) {
+        Map<Long, List<Pet>> groupPetsByMemberId = clubPetRepository.findAllByClubId(club.getId()).stream()
+                .map(ClubPet::getPet)
+                .collect(Collectors.groupingBy(pet -> pet.getMember().getId()));
+
+        return groupPetsByMemberId.values().stream()
+                .map(petList -> petList.get(0).getImageUrl().getValue())
+                .collect(toList());
+    }
```

Comment:
```suggestion
    private List<String> findPetImageUrls(Club club) {
        Map<Long, List<Pet>> groupPetsByMemberId = clubPetRepository.findAllByClubId(club.getId()).stream()
                .map(ClubPet::getPet)
                .collect(groupingBy(pet -> pet.getMember().getId()));

        return groupPetsByMemberId.values().stream()
                .map(petList -> petList.get(0).getImageUrl().getValue())
                .toList();
    }
```

1. 메서드 이름이 더 직관적이면 좋겠어요. overview는 전체 펫의 요약이라는 느낌이 들어서요.
2. 불변 리스트로 리턴하면 더 좋겠네요!